Lextenso.js

```diff
@@ -0,0 +1,182 @@
+	"label": "Lextenso",
+	"creator": "Alexandre Mimms",
+	"target": "https?://(?:www[.-])?labase[.-]lextenso[.-](?:[.-]fr)?",
+	"minVersion": "5.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2024-04-18 16:17:04"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2024 Alexandre Mimms
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+async function scrapeJournalArticle(doc, url) {
+	const references = ZU.trimInternal(text(doc, ".document-metadata-origin").replace("Issu de ", "")).split(" - ");
+	const revue = references[0];
+	const numeroRevue = references[1];
+	const page = references[2];
+	const titre = ZU.trimInternal(text(doc, "#page-title"));
+	const auteurs = doc.querySelectorAll(".document-metadata-authors-name");
+	const abstract = ZU.trimInternal(text(doc, ".cChapeau", 0));
+	const date = ZU.trimInternal(text(doc, ".document-metadata-date", 0).replace("Date de parution : ", ""));
+
+
+	let newItem = new Z.Item("journalArticle");
+	newItem.title = titre;
+	
+	for (let auteur of auteurs) {
+		const auteurNames = auteur.innerText.split(" ");
+		newItem.creators.push({
+			firstName: auteurNames[0],
+			lastName: auteurNames[1],
+			creatorType: "author",
+			fieldMode: true,
+		});
+	}
+
+	newItem.date = date;
+	newItem.abstractNote = abstract;
+	newItem.publicationTitle = revue;
+	newItem.issue = numeroRevue.replace(/n°[0]?/, "");
+	newItem.pages = page.replace(/page\s?/, "");
+	newItem.url = url;
+	newItem.language = "french";
+	newItem.complete();
+}
+
+async function scrapeBook(doc, url) {
+	// weirdly enough no real information is displayed on the book summary page, but
+	// some info, like ISBN, is shown on individual pages.
+	// So, we get the first url to one of those individual pages, then request it so we
+	// can fetch the information.
+	// I did not yet find a way to fetch the number of page or edition.
+	// I tried accessing the link of the shop, where those are displayed, but the request
+	// fails.
+	const firstItemUrl = doc.querySelectorAll(".book-summary-list li a")[0].href;
+	Z.debug(firstItemUrl);
+	const indivPage = await requestDocument(firstItemUrl);
+
+	// Accessing the metadata - reversing the list, since there can be multiple authors
```

Comment:
There might be multiple authors. In order to correctly parse the date, publisher and ISBN without knowing the number of authors beforehand, I found it easier to reverse the array, because the end will always be the same. 

e.g. : ['Author1', 'Author2', 'Date', 'Publisher', 'ISBN'] -> reversed ['ISBN', 'Publisher', 'Date', 'Author2', 'Author1'].

I then know the ISBN will always be at position 0, publisher 1, etc.